### PR TITLE
[GEARPUMP-103] Support finite stream

### PR DESCRIPTION
**Pull request by manuzhang**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the commit message is formatted like:
   `[GEARPUMP-<Jira issue #>] Meaningful description of pull request` 
 - [x] Make sure tests pass via `sbt clean test`.
 - [x] Make sure old documentation affected by the pull request has been updated and new documentation added for new functionality.

### CODE REVIEW

**Comment by manuzhang** [comment]
R: @huafengw

**Comment by manuzhang** [comment]
It took more than a minute for a simple dsl application to finish. We should improve this in follow-ups.

image

**Comment by manuzhang** [comment]
Another thing is that when the application finishes the dashboard hangs there instead of going back to the applications page. Or we should allow user to view the history metrics.

File: streaming/src/main/scala/org/apache/gearpump/streaming/task/TaskActor.scala [hunk 1]
```diff
@@ -247,7 +247,11 @@ class TaskActor(
         updateUpstreamMinClock(instant.toEpochMilli)
         minClockReported = false
       }
-      receiveMessage(watermark.toMessage, sender)
+
+      // Instant.MAX marks the end of input stream
+      if (instant.isBefore(Instant.MAX)) {
+        receiveMessage(watermark.toMessage, sender)
+      }
```
Comment (huafengw) [inline streaming/src/main/scala/org/apache/gearpump/streaming/task/TaskActor.scala]: Why the Task could not handle Instant.Max?

**Comment by huafengw** [comment]
+1

**Comment by manuzhang** [comment]
Checking minClock on `Update` instead of `HealthCheck` in ClockService reduces finish time.

**Pull request closed without merging**